File: frontend/src/app/models/difficulty.ts
export const DEFAULT_DIFFICULTY: Difficulty = { easy: 0.3, medium: 0.35, hard: 0.35, extreme: 0 };

export interface Difficulty {
  easy: number;
  medium: number;
  hard: number;
  extreme: number;
}


export function getDifficultyWord(difficulty: number): string {
  switch (difficulty) {
    case 1:
      return 'Easy';
    case 2:
      return 'Medium';
    case 3:
      return 'Hard';
    case 4:
      return 'Extreme';
    default:
      return 'None';
  }
}

File: backend/src/main/java/com/drinkster/model/enums/Difficulty.java
package com.drinkster.model.enums;

import java.util.Arrays;
import java.util.Objects;

public enum Difficulty {
    EASY("EASY"),
    MEDIUM("MEDIUM"),
    HARD("HARD"),
    EXTREME("EXTREME");

    private final String value;

    Difficulty(String value) {
        this.value = value;
    }


    public static Difficulty fromValue(String value) {
        return Arrays.stream(values())
                .filter(d -> Objects.equals(d.value, value))
                .findFirst()
                .orElseThrow(IllegalArgumentException::new);
    }
}
